capstone/src/test/java/com/google/sps/AssociationDatastoreTest.java

```diff
@@ -0,0 +1,153 @@
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  private void addComment(
+      DatastoreService datastore, String text, String zipCode, boolean associationProcessed) {
+    Entity entity = new Entity(AssociationDatastore.SURVEY_ENTITY_KIND);
+    entity.setProperty(AssociationDatastore.COMMENT_PROPERTY, text);
+    entity.setProperty(AssociationDatastore.ZIPCODE, zipCode);
+    entity.setProperty("association-processed", associationProcessed);
+    datastore.put(entity);
+  }
+
+  @Test
+  public void getCommentsEmptyTest() {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    AssociationDatastore association = new AssociationDatastore(datastore);
+    ArrayList<AssociationInput> input = association.getComments();
+    assertEquals(input, new ArrayList<AssociationInput>());
+  }
+
+  @Test
+  public void skipsProcessedEntriesTest() {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    addComment(datastore, "hi", "94123", true);
+    addComment(datastore, "test", "94124", true);
+    addComment(datastore, "testing", "94122", true);
+
+    AssociationDatastore association = new AssociationDatastore(datastore);
+    ArrayList<AssociationInput> input = association.getComments();
+    assertEquals(input, new ArrayList<AssociationInput>());
+  }
+
+  @Test
+  public void skipsProcessedReturnsNewTest() {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    addComment(datastore, "hi", "94123", true);
+    addComment(datastore, "test", "94124", false);
+    addComment(datastore, "testing", "94122", true);
+
+    AssociationDatastore association = new AssociationDatastore(datastore);
+    ArrayList<AssociationInput> input = association.getComments();
+    assertEquals(
+        input,
+        new ArrayList<AssociationInput>(
+            Arrays.asList(
+                new AssociationInput(
+                    "test", new ArrayList<String>(Arrays.asList("Bayview", "SF"))))));
+  }
+
+  public boolean isValidAssociationDatastore(ArrayList<AssociationResult> results) {
+    HashSet<String> seen = new HashSet<String>();
+    for (AssociationResult result : results) {
+      if (seen.contains(result.getContent())) {
+        return false;
+      }
+      seen.add(result.getContent());
+    }
+    return true;
+  }
+
+  @Test
+  public void addThenRetrieveTest() {
+    ArrayList<AssociationResult> data =
+        new ArrayList<AssociationResult>(
+            Arrays.asList(
+                new AssociationResult("hi", 12.4f, 118.0f, false),
+                new AssociationResult("test1", 1.3f, 123.4f, true)));
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    AssociationDatastore association = new AssociationDatastore(datastore);
+    association.storeResults(data, "SF");
+    ArrayList<AssociationResult> retrieved = association.loadPreviousResults("SF");
+    assert (isValidAssociationDatastore(retrieved));
```

Comment:
Please use JUnit's `assertTrue` or `assertFalse`

Reply:
Fixed this and future occurrences!